lib/internal/wrap_js_stream.js

```diff
@@ -126,93 +158,42 @@ class JSStreamWrap extends Socket {
 
       let errCode = 0;
       if (err) {
-        const code = uv[`UV_${err.code}`];
-        errCode = (err.code && code) ? code : uv.UV_EPIPE;
+        errCode = uv[`UV_${err.code}`] || uv.UV_EPIPE;
       }
 
       // Ensure that write was dispatched
-      setImmediate(function() {
-        // Do not invoke callback twice
-        if (!self._dequeue(item))
-          return;
-
-        handle.finishWrite(req, errCode);
+      setImmediate(() => {
+        self.finishWrite(handle, errCode);
```

Comment:
you can avoid this closure completely with `setImmediate(finishWrite, self, handle, errCode)`.